### PR TITLE
gnuradio3_8: remove

### PR DESCRIPTION
**Pull request by doronbehar**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by emilazy** [comment]
The `qradiolink` failure seems like it might relate to `spdlog` being in `-I` rather than `-isystem`, making the nested include fail.

**Comment by markuskowa** [comment]
I did not have time to take a closer look, but there seems to be a branch of qradiolink that is compatible with gr-3.10:
https://github.com/qradiolink/qradiolink/tree/gr_3.10

**Comment by doronbehar** [comment]
Thanks for the notification @emilazy . I won't mind wait until that PR reaches `master` to check again. In the meantime I resolved the merge conflicts.

**Comment by doronbehar** [comment]
OK now that `staging-next` is pretty baked, I included https://github.com/NixOS/nixpkgs/pull/365839 and I noticed I get different GR headers related errors. Then I switched qradiolink to the `gr_3.10` branch, and I only had to add `log4cpp` to the `buildInputs` and the build succeeded (upstream is probably still relying on the old log library although GR is using `spdlog` nowadays).

@markuskowa if you want to test `qradiolink` before `staging-next` reaches `master`, you should be able to simply `rebase` this PR to current `staging-next` (but don't forget to also include https://github.com/NixOS/nixpkgs/pull/365839 ). I think this PR should be ready once `staging-next` reaches `master.

**Comment by doronbehar** [comment]
Tested that qradiolink builds, launches and GUI seems functional.

**Comment by doronbehar** [comment]
staging-next was merged:

- #361878

**Comment by doronbehar** [comment]
Converted back to draft due to:

- https://github.com/xbmc/xbmc/issues/26164

Which I hope kodi developers will solve. cc @NixOS/kodi .

**Comment by doronbehar** [comment]
Fixed kodi issue.

File: pkgs/applications/radio/gnuradio/default.nix [hunk 1]
```diff
@@ -21,7 +20,7 @@
 , libjack2
 , libiio
 , libad9361
-, CoreAudio
+, apple-sdk
```
Comment (paparodeo) [inline pkgs/applications/radio/gnuradio/default.nix:23]:
```suggestion
```

File: pkgs/applications/radio/gnuradio/default.nix [hunk 2]
```diff
@@ -173,7 +172,7 @@ let
     gr-audio = {
       runtime = []
         ++ lib.optionals stdenv.hostPlatform.isLinux [ alsa-lib libjack2 ]
-        ++ lib.optionals stdenv.hostPlatform.isDarwin [ CoreAudio ]
+        ++ lib.optionals stdenv.hostPlatform.isDarwin [ apple-sdk ]
```
Comment (paparodeo) [inline pkgs/applications/radio/gnuradio/default.nix:175]:
```suggestion
```

File: pkgs/development/gnuradio-modules/osmosdr/default.nix [hunk 3]
```diff
@@ -1,6 +1,6 @@
 { lib
 , stdenv
-, darwin
+, apple-sdk
```
Comment (paparodeo) [inline pkgs/development/gnuradio-modules/osmosdr/default.nix:3]:
```suggestion
```

File: pkgs/development/gnuradio-modules/osmosdr/default.nix [hunk 4]
```diff
@@ -72,8 +61,7 @@ in mkDerivation {
       python.pkgs.numpy
       python.pkgs.pybind11
   ] ++ lib.optionals stdenv.hostPlatform.isDarwin [
-    darwin.apple_sdk.frameworks.IOKit
-    darwin.apple_sdk.frameworks.Security
+    apple-sdk
```
Comment (paparodeo) [inline pkgs/development/gnuradio-modules/osmosdr/default.nix:64]:
```suggestion
```

**Review by paparodeo — COMMENTED** [review]
LGTM -- don't need to add `apple-sdk` as that comes with the stdenv. if you need an sdk other than the default you'd do something like `apple-sdk_15` which would bump the sdk to version 15 rather than 11, the current default for both platforms.

**Comment by doronbehar** [comment]
Thanks for the review @paparodeo :). Done.

**Review by paparodeo — APPROVED** [review]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 356976`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>gnuradio</li>
    <li>gnuradioMinimal</li>
    <li>gnuradioPackages.osmosdr.dev</li>
    <li>gnuradioPackages.osmosdr</li>
    <li>trunk-recorder</li>
  </ul>
</details>

**Comment by doronbehar** [comment]
Friendly ping @markuskowa

**Review by markuskowa — APPROVED** [review]
Thanks for fixing everything.

**Comment by emilazy** [comment]
Thanks for tackling this!

**Pull request merged**